src/ngraph/frontend/onnxifi/tensor.hpp

```diff
@@ -0,0 +1,141 @@
+
+            Tensor() = delete;
+            virtual ~Tensor() = default;
+
+            explicit Tensor(const ::onnxTensorDescriptorV1& tensor)
+                : m_tensor{&tensor}
+            {
+                if (tensor.tag != ONNXIFI_TAG_TENSOR_DESCRIPTOR_V1)
+                {
+                    throw status::unsupported_tag{};
+                }
+                if (tensor.name == nullptr)
+                {
+                    throw status::invalid_name{};
+                }
+                switch (tensor.dataType)
+                {
+                case ONNXIFI_DATATYPE_FLOAT16:
+                case ONNXIFI_DATATYPE_FLOAT32:
+                case ONNXIFI_DATATYPE_FLOAT64:
+                case ONNXIFI_DATATYPE_INT8:
+                case ONNXIFI_DATATYPE_INT16:
+                case ONNXIFI_DATATYPE_INT32:
+                case ONNXIFI_DATATYPE_INT64:
+                case ONNXIFI_DATATYPE_UINT8:
+                case ONNXIFI_DATATYPE_UINT16:
+                case ONNXIFI_DATATYPE_UINT32:
+                case ONNXIFI_DATATYPE_UINT64: break;
+                case ONNXIFI_DATATYPE_COMPLEX64:
+                case ONNXIFI_DATATYPE_COMPLEX128: throw status::invalid_datatype{};
+                default: throw status::unsupported_datatype{};
+                }
+                switch (tensor.memoryType)
+                {
+                case ONNXIFI_MEMORY_TYPE_CPU: break;
+                case ONNXIFI_MEMORY_TYPE_CUDA_BUFFER:
+                case ONNXIFI_MEMORY_TYPE_OPENCL_BUFFER:
+                case ONNXIFI_MEMORY_TYPE_OPENGLES_TEXTURE_2D:
+                case ONNXIFI_MEMORY_TYPE_D3D_RESOURCE: throw status::invalid_memory_type{};
+                default: throw status::unsupported_memory_type{};
+                }
+                if ((tensor.dimensions != 0) && (tensor.shape == nullptr))
+                {
+                    throw status::null_pointer{};
+                }
+                if ((tensor.shape != nullptr) && (tensor.dimensions == 0))
+                {
+                    throw status::invalid_size{};
+                }
+                if (tensor.shape == nullptr)
+                {
+                    m_shape = {1};
+                }
+                else
+                {
+                    Span<uint64_t> shape{tensor.shape, tensor.dimensions};
+                    for (const auto& value : shape)
+                    {
+                        if (value == 0)
+                        {
+                            throw status::invalid_shape{};
+                        }
+                        m_shape.push_back(value);
+                        m_size *= value;
+                    }
+                }
+                if (tensor.buffer == 0)
+                {
+                    throw status::invalid_memory_location{};
+                }
+            }
+
+            virtual std::shared_ptr<runtime::Tensor> to_ng(runtime::Backend& backend) const = 0;
```

Comment:
Please add a line or two of comments for this function.

Reply:
done